src/transport/reqwest.rs

```diff
@@ -189,6 +189,267 @@ impl PubNubClientBuilder<TransportReqwest> {
+    //!
+    //! It requires the [`reqwest` and `blocking` feature] to be enabled.
+    //!
+    //! [`TransportReqwest`]: ./struct.TransportReqwest.html
+    //! [`PubNub API`]: https://www.pubnub.com/docs
+    //! [`reqwest`]: https://docs.rs/reqwest
+    //! [`pubnub`]: ../index.html
+    //! [`PubNubClient`]: ../pubnub_client/struct.PubNubClient.html
+    //! [`reqwest` feature]: ../index.html#features
+
+    use log::info;
+
+    use crate::{
+        core::{PubNubError, TransportMethod, TransportRequest, TransportResponse},
+        transport::reqwest::{prepare_headers, prepare_url},
+        PubNubClientBuilder,
+    };
+
+    /// This struct is used to send requests to the [`PubNub API`] using the [`reqwest`] crate.
+    /// It is used as the transport type for the [`PubNubClient`].
+    /// It is intended to be used by the [`pubnub`] crate.
+    ///
+    /// It requires the [`reqwest` and `blocking` feature] to be enabled.
+    ///
+    /// [`PubNub API`]: https://www.pubnub.com/docs
+    /// [`reqwest`]: https://docs.rs/reqwest
+    /// [`pubnub`]: ../index.html
+    /// [`PubNubClient`]: ../pubnub_client/struct.PubNubClient.html
+    pub struct TransportReqwest {
+        reqwest_client: reqwest::blocking::Client,
+        /// The hostname to use for requests.
+        /// It is used as the base URL for all requests.
+        ///
+        /// It defaults to `https://ps.pndsn.com/`.
+        /// # Examples
+        /// ```
+        /// use pubnub::transport::TransportReqwest;
+        ///
+        /// let transport = {
+        ///    let mut transport = TransportReqwest::default();
+        ///    transport.hostname = "https://wherever.you.want.com/".into();
+        ///    transport
+        /// };
+        /// ```
+        pub hostname: String,
+    }
+
+    impl crate::core::blocking::Transport for TransportReqwest {
+        fn send(&self, request: TransportRequest) -> Result<TransportResponse, PubNubError> {
+            let request_url = prepare_url(&self.hostname, &request.path, &request.query_parameters);
+            info!("{}", request_url);
+            let headers = prepare_headers(&request.headers)?;
+            let builder = match request.method {
+                TransportMethod::Get => self.prepare_get_method(request, request_url),
+                TransportMethod::Post => self.prepare_post_method(request, request_url),
+            }?;
+
+            let result = builder
+                .headers(headers)
+                .send()
+                .map_err(|e| PubNubError::TransportError(e.to_string()))?;
+
+            Ok(TransportResponse {
+                status: result.status().as_u16(),
+                body: result
+                    .bytes()
+                    .map(|b| b.to_vec())
+                    .map(|b| (!b.is_empty()).then_some(b))
+                    .map_err(|e| PubNubError::TransportError(e.to_string()))?,
+                ..Default::default()
+            })
```

Comment:
It's not copied at all. 
It operates at totally different types that are named in the same way in `reqwest` crate ;/

I can try to abstract some of this things but I'm afraid that it will be harder to understand.